src/runtime/ExtendedNativeFunctionObject.h

```diff
@@ -0,0 +1,31 @@
+#ifndef __EscargotExtendedNativeFunctionObject__
+#define __EscargotExtendedNativeFunctionObject__
+
+#include "NativeFunctionObject.h"
+
+namespace Escargot {
+
+class ExtendedNativeFunctionObject : public NativeFunctionObject {
+public:
+    ExtendedNativeFunctionObject(ExecutionState& state, NativeFunctionInfo info, std::vector<Value> values);
+
+    virtual bool isExtendedNativeFunctionObject() const
+    {
+        return true;
+    }
+
+    std::vector<Value> getInternalValues()
+    {
+        return m_values;
+    }
+
+    void* operator new(size_t size);
+    void* operator new[](size_t size) = delete;
+
+
+private:
+    std::vector<Value> m_values;
```

Comment:
Unfortunately, our GC could not recognize the STL vector correctly. Please replace it with `Escargot::Vector`. One more thing, Escargot generally uses `SmallValue` form when storing `Value` in object (conversion overhead between `Value` and `SmallValue` is added, but SmallValue consumes less memory compared to Value). In this case, `SmallValueVector` would be suitable.